src/fu-engine.c

```diff
@@ -5017,6 +5020,188 @@ fu_engine_get_host_machine_id (FuEngine *self)
 }
 
+static void
+fu_engine_add_hsi_attrs_tainted (FuEngine *self, GPtrArray *attrs)
+{
+	FwupdHsiAttr *attr;
+
+	/* everything is fine */
+	if (!self->tainted)
+		return;
+
+	/* add runtime attribute */
+	attr = fwupd_hsi_attr_new ("org.fwupd.Hsi.PluginsTainted");
+	fwupd_hsi_attr_set_name (attr, "fwupd daemon");
+	fwupd_hsi_attr_set_summary (attr, "The daemon has been tainted by a non-upstream plugin");
+	fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_RUNTIME_ISSUE);
+	g_ptr_array_add (attrs, attr);
+}
+
+static void
+fu_engine_add_hsi_attrs_supported (FuEngine *self, GPtrArray *attrs)
+{
+	FwupdHsiAttr *attr;
+	FwupdRelease *rel_newest = NULL;
+	guint64 now = (guint64) g_get_real_time () / G_USEC_PER_SEC;
+	g_autoptr(FuDevice) device = NULL;
+	g_autoptr(GPtrArray) releases = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* find out if there is firmware less than 6 months old */
+	device = fu_device_list_get_by_guid (self->device_list,
+					     /* main-system-firmware */
+					     "230c8b18-8d9b-53ec-838b-6cfc0383493a",
+					     &error_local);
+	if (device == NULL) {
+		g_debug ("no device for main-system-firmware: %s",
+			 error_local->message);
+		return;
+	}
+	releases = fu_engine_get_releases_for_device (self, device, &error_local);
+	if (releases == NULL) {
+		g_debug ("no releases for main-system-firmware: %s",
+			 error_local->message);
+		return;
+	}
+
+	/* check the age */
+	for (guint i = 0; i < releases->len; i++) {
+		FwupdRelease *rel_tmp = g_ptr_array_index (releases, i);
+		if (rel_newest == NULL ||
+		    fwupd_release_get_created (rel_tmp) > fwupd_release_get_created (rel_newest))
+			rel_newest = rel_tmp;
+	}
+	if (now - fwupd_release_get_created (rel_newest) > 60 * 60 * 24 * 30 * 12) {
+		g_debug ("newest release is %" G_GUINT64_FORMAT " months old",
+			 (now - fwupd_release_get_created (rel_newest)) / (60 * 60 * 24 * 30));
+		return;
+	}
+
+	/* add runtime attribute */
+	attr = fwupd_hsi_attr_new ("org.fwupd.Hsi.Updates");
+	fwupd_hsi_attr_set_name (attr, "Firmware Supported");
+	fwupd_hsi_attr_set_summary (attr, "There is firmware less than 12 months old");
+	fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_SUCCESS);
+	fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_RUNTIME_UPDATES);
+	g_ptr_array_add (attrs, attr);
+
+	/* do we have attestation checksums */
+	if (fwupd_release_get_checksums(rel_newest)->len > 0) {
+		attr = fwupd_hsi_attr_new ("org.fwupd.Hsi.Attestation");
+		fwupd_hsi_attr_set_name (attr, "Firmware Attestation");
+		fwupd_hsi_attr_set_summary (attr, "Firmware has attestation checksums");
+		fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_SUCCESS);
+		fwupd_hsi_attr_add_flag (attr, FWUPD_HSI_ATTR_FLAG_RUNTIME_ATTESTATION);
+		g_ptr_array_add (attrs, attr);
+	}
+}
+
+GPtrArray *
+fu_engine_get_host_security_attrs (FuEngine *self, GError **error)
+{
+	GPtrArray *plugins = fu_plugin_list_get_all (self->plugin_list);
+	g_autoptr(GPtrArray) hsi_attrs = NULL;
+
```

Comment:
I think you want two extra checks here:
1. At least the way that HSI is done right now, should make sure the system is booted in UFEI mode and return an error if not.
2. Check that it's x86_64.  Most of the architecture for HSI is x86 centered.

Reply:
> make sure the system is booted in UFEI mode

I don't think HSI needs to be x86 centered, I just think for the other architectures we return HSI:0 until we define some sane firmware platform tests.

Reply:
Well several things are part of the x86 PC architecture specifically and not really seen elsewhere: TPM, EC.  Things like the IOMMU are seen in different implementations on other architectures.

Reply:
And those PCH bits in HSI:1 are x86 specific too.

Reply:
> make sure the system is booted in UFEI mode

What's the best way to check if we're in EFI mode? Check that /sys/firmware/efi/runtime exists?

Reply:
That's what we do in the UEFI plugin: https://github.com/fwupd/fwupd/blob/0258c12af38eae27812695d3351a3797b65b8d6c/plugins/uefi/fu-plugin-uefi.c#L607

Maybe that should be generalized somewhere else now in libfwupdplugin?